### PR TITLE
File renamed/replaced whitelist with allowlist

### LINKED ISSUE #41748: Replace blacklist/whitelist in cmake/Whitelist.cmake
The goal of this issue is to replace all instances of *whitelist* and *blacklist* in `cmake/Whitelist.cmake`. See issue #41443 for more information.

### PR DESCRIPTION
**Pull request by kalmufti**
Fixes #41748

Renamed the source file from Whitelist.cmake to Allowlist.cmake and replaced all whitelist occurrences within the source with allowlist.

### CODE REVIEW

**Comment by ngimel** [comment]
CI failures are real.

**Review by SplitInfinity — CHANGES_REQUESTED** [review]
Please fix the CI errors.

**Comment by kalmufti** [comment]
I've fixed the include reference in pytorch/CMakeLists.txt for the new filename, though I didn't change all the whitelist references since it would've caused a different CI errors. It seems that now it passes the previous error but I'm not sure what this new one is. I'd be happy to fix it, if you can give me some guidance.

**Review by SplitInfinity — APPROVED** [review]
Thanks for working on this!

**Comment by kalmufti** [comment]
My pleasure thanks! Glad to be part of such great project.

File: cmake/Allowlist.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+endif()
+
+# First read the allowlist file and break it by line.
+file(READ "${CAFFE2_ALLOWLIST}" allowlist_content)
+# Convert file contents into a CMake list
+string(REGEX REPLACE "\n" ";" allowlist_content ${allowlist_content})
+
+foreach(item ${allowlist_content})
+  file(GLOB_RECURSE tmp ${item})
+  set(CAFFE2_ALLOWLISTED_FILES ${CAFFE2_ALLOWLISTED_FILES} ${tmp})
```
Comment (ljk53) [inline cmake/Allowlist.cmake:20]: You probably need also modify here: https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L147, right?

File: cmake/Allowlist.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
+
+if(__caffe2_allowlist_included)
+  return()
+endif()
+
+set(__caffe2_allowlist_included TRUE)
+
+set(CAFFE2_ALLOWLISTED_FILES)
+if(NOT CAFFE2_ALLOWLIST)
```
Comment (ljk53) [inline cmake/Allowlist.cmake:9]: I think this should be consistent with here: https://github.com/pytorch/pytorch/blob/master/CMakeLists.txt#L452

I don't know how/whether this is used by any other OSS projects - looks it's introduced for caffe2 selective build: https://github.com/facebookarchive/caffe2/issues/285. Updating external visible flags probably should be mentioned in release note. cc: @dzhulgakov  who might have more historical context.

Comment (SplitInfinity) [reply cmake/Allowlist.cmake:9]: If this is used by external projects, let's not modify it for now. Once all of the easy replacements are done, regular contributors (e.g. like me) can handle more complicated cases like this that have dependencies on other projects. I don't think such task is appropriate for new or first-time contributors.

**Comment by ljk53** [comment]
Hi @kalmufti - has this PR already been merged into #42067 ? Is there anything else to be added on this one?

**Comment by kalmufti** [comment]
@ljk53, think is't ready, though there is still one pending reviewer.

**Comment by ljk53** [comment]
> @ljk53, think is't ready, though there is still one pending reviewer.

I mean this PR seems to become empty (you made the change in 42067 instead) - did I miss something? :)

**Comment by kalmufti** [comment]
Oh I see 😄 guess we don't need this anymore. On the other PR, I was trying to make sure it passes all checks, that's why I had to make the changes in that one as well.

**Comment by julohaq22** [comment]
> Fixes #41748
> 
> Renamed the source file from Whitelist.cmake to Allowlist.cmake and replaced all whitelist occurrences within the source with allowlist.

**Pull request closed without merging**